src/figuresClasses.ts

```diff
@@ -1,19 +1,111 @@
-export interface Figure {
 
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  public color: Color;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    const sidesArr = [a, b, c];
+
+    sidesArr.sort((x:number, y:number):number => (x > y ? -1 : 1));
+
+    if (sidesArr[0]
+      >= (sidesArr.reduce((x:number, y:number) => x + y, 0) - sidesArr[0])) {
+      throw new Error('your error message');
```

Comment:
Replace 'your error message' with a meaningful error message, such as 'Longest side of the triangle must be smaller than the sum of the other two sides.'